Extensions/TweenBehavior/tweenruntimebehavior.ts

```diff
@@ -104,16 +116,10 @@ namespace gdjs {
     }
 
     onDestroy() {
-      const shiftyJsScene = this._runtimeScene.shiftyJsScene;
-      if (!shiftyJsScene) return;
-
       // Stop and delete all tweens of the behavior - otherwise they could:
       // - continue to point to the behavior, and so to the object (memory leak),
       // - affect the object in case it's recycled (wrong/hard to debug behavior).
-      for (const identifier in this._tweens) {
-        this._tweens[identifier].instance.stop();
-        shiftyJsScene.remove(this._tweens[identifier].instance);
-      }
+      for (const identifier in this._tweens) this.removeTween(identifier);
```

Comment:
Is there a risk of memory leak? How long is the object/behavior still leaving in memory after it's removed from the scene?

Reply:
At the beginning of the next frame, since removeTween calls hasFinished on the tween, the AsyncTasksManager should run the callback and delete the task, which has the reference to the tween which likely captured the RuntimeObject with its step function. 

The only risk of memory leaks because of that is if the object is somehow deleted without `onDestroy` being called, but that will most likely never happen.

Reply:
I have bad gut feeling about this - didn't we had an issue about calling removeTween at some wrong point and we had to fix it but it was hard to debug?

I wonder if tasks awaiting should somehow be notified immediately that the object is destroyed and so they are considered as resolved (at the next tick sure, but so that they can drop the reference to the tween immediately).

What I mean is that the object instances are "hold" by the runtime scene, who is the owner of them, and objects own their behavior. When objects are destroyed, they notify their behaviors that they are being destroyed (onDestroy).
While the runtime scene can maybe decide to keep or not objects/behaviors in a cache to recycle them, it probably won't and so they should be removed from memory. In a language like C++, we would probably `delete` the objects.

But there, because the tasks have a reference to the tween, they force the behavior to stay in memory and be GC'ed later. It's a bit going in the wrong direction. It means a task could do stuff with the tween, while it's deleted. In C++, tasks could keep a weak pointer. In JS we can't, it's not a big deal, but we should still strive to avoid keeping references to things that are logically destroyed.
Otherwise, this liberty that we're taking is going to bite us one day when a task will be updated to check something else, that will call something else, that will end up modify a behavior that was supposed to be destroyed and still has somehow a side effect.

I'm not sure what the good solution is exactly, but tasks awaiting on a tween should probably be notified, so they are "aborted" or "force resolved" in a way.

Reply:
> In C++, tasks could keep a weak pointer.

We can, actually, but Opera doesn't support it yet: https://caniuse.com/mdn-javascript_builtins_weakref

Reply:
Is this satisfactory? 0322e0ecd9ed9c3b70c0eb34e137c9486ab0c96c
Or have I misunderstood what you wanted?

Reply:
Yes I think that was the overall idea 👍 Be 100% sure that the lifecycle of objects/tween behavior/tasks is precise and consistently handled.
I think I saw a leak of the tasks, I will add a comment.